diff -Naur elfutils-0.191-orig/lib/eu-config.h elfutils-0.191/lib/eu-config.h
--- elfutils-0.191-orig/lib/eu-config.h	2024-08-09 04:44:22.337853677 +0000
+++ elfutils-0.191/lib/eu-config.h	2024-08-09 04:44:45.017990253 +0000
@@ -62,7 +62,7 @@
 #include <libintl.h>
 /* gettext helper macros.  */
 #define N_(Str) Str
-#define _(Str) dgettext ("elfutils", Str)
+/* #define _(Str) dgettext ("elfutils", Str) */
 
 /* Compiler-specific definitions.  */
 #define strong_alias(name, aliasname) \
