diff -Naur libdbusmenu-qt-0.9.3+16.04.20160218-orig/src/dbusmenuexporter.cpp libdbusmenu-qt-0.9.3+16.04.20160218/src/dbusmenuexporter.cpp
--- libdbusmenu-qt-0.9.3+16.04.20160218-orig/src/dbusmenuexporter.cpp	2024-09-12 17:15:47.642787276 -0700
+++ libdbusmenu-qt-0.9.3+16.04.20160218/src/dbusmenuexporter.cpp	2024-09-12 17:16:12.466471478 -0700
@@ -29,6 +29,7 @@
 #include <QTimer>
 #include <QToolButton>
 #include <QWidgetAction>
+#include <QActionGroup>
 
 // Local
 #include "dbusmenu_config.h"
@@ -484,7 +485,7 @@
 {
     int id = d->idForAction(action);
     DMRETURN_IF_FAIL(id >= 0);
-    uint timeStamp = QDateTime::currentDateTime().toTime_t();
+    uint timeStamp = QDateTime::currentDateTime().toSecsSinceEpoch();
     d->m_dbusObject->ItemActivationRequested(id, timeStamp);
 }
 
diff -Naur libdbusmenu-qt-0.9.3+16.04.20160218-orig/src/dbusmenuimporter.cpp libdbusmenu-qt-0.9.3+16.04.20160218/src/dbusmenuimporter.cpp
--- libdbusmenu-qt-0.9.3+16.04.20160218-orig/src/dbusmenuimporter.cpp	2024-09-12 17:15:46.906796640 -0700
+++ libdbusmenu-qt-0.9.3+16.04.20160218/src/dbusmenuimporter.cpp	2024-09-12 17:19:38.383851779 -0700
@@ -34,6 +34,7 @@
 #include <QTimer>
 #include <QToolButton>
 #include <QWidgetAction>
+#include <QActionGroup>
 
 // Local
 #include "dbusmenutypes_p.h"
@@ -338,7 +339,7 @@
 
     d->m_type = type;
 
-    connect(&d->m_mapper, SIGNAL(mapped(int)), SLOT(sendClickedEvent(int)));
+    connect(&d->m_mapper, SIGNAL(mappedInt(int)), SLOT(sendClickedEvent(int)));
 
     d->m_pendingLayoutUpdateTimer = new QTimer(this);
     d->m_pendingLayoutUpdateTimer->setSingleShot(true);
diff -Naur libdbusmenu-qt-0.9.3+16.04.20160218-orig/src/dbusmenushortcut_p.cpp libdbusmenu-qt-0.9.3+16.04.20160218/src/dbusmenushortcut_p.cpp
--- libdbusmenu-qt-0.9.3+16.04.20160218-orig/src/dbusmenushortcut_p.cpp	2024-09-12 17:15:46.794798065 -0700
+++ libdbusmenu-qt-0.9.3+16.04.20160218/src/dbusmenushortcut_p.cpp	2024-09-12 17:16:12.570470155 -0700
@@ -22,6 +22,7 @@
 
 // Qt
 #include <QtGui/QKeySequence>
+#include <QtDBus/QDBusArgument>
 
 // Local
 #include "debug_p.h"
@@ -83,3 +84,29 @@
     QString string = tmp.join(QLatin1String(", "));
     return QKeySequence::fromString(string);
 }
+
+QDBusArgument &operator<<(QDBusArgument &argument, const DBusMenuShortcut &obj)
+{
+    int id = qMetaTypeId<QStringList>();
+    argument.beginArray(id);
+    typename DBusMenuShortcut::ConstIterator it = obj.constBegin();
+    typename DBusMenuShortcut::ConstIterator end = obj.constEnd();
+    for ( ; it != end; ++it)
+        argument << *it;
+    argument.endArray();
+    return argument;
+}
+
+const QDBusArgument &operator>>(const QDBusArgument &argument, DBusMenuShortcut &obj)
+{
+    argument.beginArray();
+    obj.clear();
+    while (!argument.atEnd()) {
+        QStringList item;
+        argument >> item;
+        obj.push_back(item);
+    }
+    argument.endArray();
+
+    return argument;
+}
diff -Naur libdbusmenu-qt-0.9.3+16.04.20160218-orig/src/dbusmenushortcut_p.h libdbusmenu-qt-0.9.3+16.04.20160218/src/dbusmenushortcut_p.h
--- libdbusmenu-qt-0.9.3+16.04.20160218-orig/src/dbusmenushortcut_p.h	2024-09-12 17:15:47.822784987 -0700
+++ libdbusmenu-qt-0.9.3+16.04.20160218/src/dbusmenushortcut_p.h	2024-09-12 17:16:12.618469544 -0700
@@ -30,6 +30,7 @@
 
 
 class QKeySequence;
+class QDBusArgument;
 
 class DBUSMENU_EXPORT DBusMenuShortcut : public QList<QStringList>
 {
@@ -40,4 +41,7 @@
 
 Q_DECLARE_METATYPE(DBusMenuShortcut)
 
+DBUSMENU_EXPORT QDBusArgument &operator<<(QDBusArgument &argument, const DBusMenuShortcut &);
+DBUSMENU_EXPORT const QDBusArgument &operator>>(const QDBusArgument &argument, DBusMenuShortcut &);
+
 #endif /* DBUSMENUSHORTCUT_H */
