diff -Naur hollywood-desktop-1.0/desktop/compositor/include/appmenu.h hollywood-desktop-1.0-qt65/desktop/compositor/include/appmenu.h
--- hollywood-desktop-1.0/desktop/compositor/include/appmenu.h	2023-03-12 18:24:41.000000000 -0700
+++ hollywood-desktop-1.0-qt65/desktop/compositor/include/appmenu.h	2023-08-23 17:42:30.282454072 -0700
@@ -6,7 +6,7 @@
 #define APPMENU_H
 
 #include <QWaylandCompositorExtensionTemplate>
-#include <QWaylandQuickExtension>
+//#include <QWaylandQuickExtension>
 #include <QWaylandCompositor>
 
 #include "qwayland-server-appmenu.h"
diff -Naur hollywood-desktop-1.0/desktop/compositor/include/fullscreen.h hollywood-desktop-1.0-qt65/desktop/compositor/include/fullscreen.h
--- hollywood-desktop-1.0/desktop/compositor/include/fullscreen.h	2023-03-12 18:24:41.000000000 -0700
+++ hollywood-desktop-1.0-qt65/desktop/compositor/include/fullscreen.h	2023-08-23 17:42:30.282454072 -0700
@@ -7,7 +7,7 @@
 
 #include <QObject>
 #include <QWaylandCompositorExtensionTemplate>
-#include <QWaylandQuickExtension>
+//#include <QWaylandQuickExtension>
 #include <QWaylandCompositor>
 #include <QWaylandResource>
 
diff -Naur hollywood-desktop-1.0/desktop/compositor/include/gtkshell.h hollywood-desktop-1.0-qt65/desktop/compositor/include/gtkshell.h
--- hollywood-desktop-1.0/desktop/compositor/include/gtkshell.h	2023-03-12 18:24:41.000000000 -0700
+++ hollywood-desktop-1.0-qt65/desktop/compositor/include/gtkshell.h	2023-08-23 17:42:30.282454072 -0700
@@ -6,7 +6,7 @@
 #define GTKSHELL_H
 
 #include <QWaylandCompositorExtensionTemplate>
-#include <QWaylandQuickExtension>
+//#include <QWaylandQuickExtension>
 #include <QWaylandCompositor>
 #include <QWaylandResource>
 
diff -Naur hollywood-desktop-1.0/desktop/compositor/include/qtshell.h hollywood-desktop-1.0-qt65/desktop/compositor/include/qtshell.h
--- hollywood-desktop-1.0/desktop/compositor/include/qtshell.h	2023-03-12 18:24:41.000000000 -0700
+++ hollywood-desktop-1.0-qt65/desktop/compositor/include/qtshell.h	2023-08-23 17:42:30.282454072 -0700
@@ -6,7 +6,7 @@
 #define QTSHELL_H
 
 #include <QWaylandCompositorExtensionTemplate>
-#include <QWaylandQuickExtension>
+//#include <QWaylandQuickExtension>
 #include <QWaylandCompositor>
 #include <QWaylandResource>
 
diff -Naur hollywood-desktop-1.0/desktop/compositor/include/screencopy.h hollywood-desktop-1.0-qt65/desktop/compositor/include/screencopy.h
--- hollywood-desktop-1.0/desktop/compositor/include/screencopy.h	2023-03-12 18:24:41.000000000 -0700
+++ hollywood-desktop-1.0-qt65/desktop/compositor/include/screencopy.h	2023-08-23 17:42:30.282454072 -0700
@@ -7,7 +7,7 @@
 
 #include <QObject>
 #include <QWaylandCompositorExtensionTemplate>
-#include <QWaylandQuickExtension>
+//#include <QWaylandQuickExtension>
 #include <QWaylandCompositor>
 #include <QWaylandResource>
 
diff -Naur hollywood-desktop-1.0/desktop/compositor/include/wndmgmt.h hollywood-desktop-1.0-qt65/desktop/compositor/include/wndmgmt.h
--- hollywood-desktop-1.0/desktop/compositor/include/wndmgmt.h	2023-03-12 18:24:41.000000000 -0700
+++ hollywood-desktop-1.0-qt65/desktop/compositor/include/wndmgmt.h	2023-08-23 17:42:30.282454072 -0700
@@ -8,7 +8,7 @@
 #include <QUuid>
 
 #include <QWaylandCompositorExtensionTemplate>
-#include <QWaylandQuickExtension>
+//#include <QWaylandQuickExtension>
 #include <QWaylandCompositor>
 #include <QIcon>
 
diff -Naur hollywood-desktop-1.0/desktop/compositor/include/xdgshell.h hollywood-desktop-1.0-qt65/desktop/compositor/include/xdgshell.h
--- hollywood-desktop-1.0/desktop/compositor/include/xdgshell.h	2023-03-12 18:24:41.000000000 -0700
+++ hollywood-desktop-1.0-qt65/desktop/compositor/include/xdgshell.h	2023-08-23 17:42:30.282454072 -0700
@@ -11,7 +11,7 @@
 #include <QtWaylandCompositor/QWaylandShell>
 #include <QtWaylandCompositor/QWaylandShellSurface>
 #include <QtWaylandCompositor/QWaylandSurface>
-#include <QtWaylandCompositor/qwaylandquickchildren.h>
+//#include <QtWaylandCompositor/qwaylandquickchildren.h>
 #include <QtWaylandCompositor/private/qwaylandcompositorextension_p.h>
 #include <QtWaylandCompositor/private/qwaylandshell_p.h>
 #include <QtWaylandCompositor/private/qwayland-server-xdg-shell.h>
@@ -83,7 +83,7 @@
     Q_INVOKABLE void initialize(HWWaylandXdgShell* xdgShell, QWaylandSurface *surface, const QWaylandResource &resource);
 
     Qt::WindowType windowType() const override;
-    QWaylandQuickShellIntegration *createIntegration(QWaylandQuickShellSurfaceItem *item) override;
+//    QWaylandQuickShellIntegration *createIntegration(QWaylandQuickShellSurfaceItem *item) override;
 
     HWWaylandXdgShell *shell() const;
     QWaylandSurface *surface() const;
diff -Naur hollywood-desktop-1.0/desktop/compositor/include/xdgshell_p.h hollywood-desktop-1.0-qt65/desktop/compositor/include/xdgshell_p.h
--- hollywood-desktop-1.0/desktop/compositor/include/xdgshell_p.h	2023-03-12 18:24:41.000000000 -0700
+++ hollywood-desktop-1.0-qt65/desktop/compositor/include/xdgshell_p.h	2023-08-23 17:42:30.282454072 -0700
@@ -10,7 +10,7 @@
 #include <QtWaylandCompositor/QWaylandResource>
 #include <QtWaylandCompositor/QWaylandShell>
 #include <QtWaylandCompositor/QWaylandShellSurface>
-#include <QtWaylandCompositor/qwaylandquickchildren.h>
+//#include <QtWaylandCompositor/qwaylandquickchildren.h>
 #include <QtWaylandCompositor/private/qwaylandcompositorextension_p.h>
 #include <QtWaylandCompositor/private/qwaylandshell_p.h>
 #include <QtWaylandCompositor/private/qwayland-server-xdg-shell.h>
diff -Naur hollywood-desktop-1.0/desktop/compositor/src/xdgshell.cc hollywood-desktop-1.0-qt65/desktop/compositor/src/xdgshell.cc
--- hollywood-desktop-1.0/desktop/compositor/src/xdgshell.cc	2023-03-12 18:24:41.000000000 -0700
+++ hollywood-desktop-1.0-qt65/desktop/compositor/src/xdgshell.cc	2023-08-23 17:42:30.292454072 -0700
@@ -660,11 +660,11 @@
     return nullptr;
 }
 
-QWaylandQuickShellIntegration *HWWaylandXdgSurface::createIntegration(QWaylandQuickShellSurfaceItem *item)
+/* QWaylandQuickShellIntegration *HWWaylandXdgSurface::createIntegration(QWaylandQuickShellSurfaceItem *item)
 {
     Q_UNUSED(item)
     return nullptr;
-}
+} */
 
 
 /*!
diff -Naur hollywood-desktop-1.0/desktop/eglfs/plugins/platforms/eglfs/api/qeglfswindow.cpp hollywood-desktop-1.0-qt65/desktop/eglfs/plugins/platforms/eglfs/api/qeglfswindow.cpp
--- hollywood-desktop-1.0/desktop/eglfs/plugins/platforms/eglfs/api/qeglfswindow.cpp	2023-03-12 18:24:41.000000000 -0700
+++ hollywood-desktop-1.0-qt65/desktop/eglfs/plugins/platforms/eglfs/api/qeglfswindow.cpp	2023-08-23 17:42:30.282454072 -0700
@@ -145,7 +145,11 @@
         m_rasterCompositingContext->setScreen(window()->screen());
         if (Q_UNLIKELY(!m_rasterCompositingContext->create()))
             qFatal("EGLFS: Failed to create compositing context");
+#if QT_VERSION < 0x060500
         compositor->setTarget(m_rasterCompositingContext, window(), screen->rawGeometry());
+#else
+	compositor->setTargetWindow(window(), screen->rawGeometry());
+#endif
         compositor->setRotation(qEnvironmentVariableIntValue("QT_QPA_EGLFS_ROTATION"));
         // If there is a "root" window into which raster and QOpenGLWidget content is
         // composited, all other contexts must share with its context.
@@ -155,6 +159,10 @@
             // AA_ShareOpenGLContexts. Set the attribute to be fully consistent.
             QCoreApplication::setAttribute(Qt::AA_ShareOpenGLContexts);
         }
+#if QT_VERSION > 0x060500
+	QOpenGLCompositor *compositor = QOpenGLCompositor::instance();
+	    compositor->setTargetContext(m_rasterCompositingContext);
+#endif
     }
 #endif // QT_NO_OPENGL
 }
diff -Naur hollywood-desktop-1.0/desktop/elevator/org.originull.elevator.desktop hollywood-desktop-1.0-qt65/desktop/elevator/org.originull.elevator.desktop
--- hollywood-desktop-1.0/desktop/elevator/org.originull.elevator.desktop	1969-12-31 16:00:00.000000000 -0800
+++ hollywood-desktop-1.0-qt65/desktop/elevator/org.originull.elevator.desktop	2023-08-23 17:42:30.312454072 -0700
@@ -0,0 +1,8 @@
+[Desktop Entry]
+Type=Application
+Exec=/usr/libexec/hollywood/elevator
+Name=Elevator
+Terminal=False
+Icon=preferences-security
+NoDisplay=true
+X-Hollywood-System=true
diff -Naur hollywood-desktop-1.0/desktop/platformtheme/include/platformtheme.h hollywood-desktop-1.0-qt65/desktop/platformtheme/include/platformtheme.h
--- hollywood-desktop-1.0/desktop/platformtheme/include/platformtheme.h	2023-03-12 18:24:41.000000000 -0700
+++ hollywood-desktop-1.0-qt65/desktop/platformtheme/include/platformtheme.h	2023-08-23 17:42:30.262454072 -0700
@@ -57,7 +57,7 @@
     void globalMenuBarNoLongerExists();
     void windowCreated(QWindow *window);
 
-#if QT_VERSION > 0x060000
+#if QT_VERSION > 0x060000 && QT_VERSION < 0x060500
     Appearance appearance() const override;
 #endif
     const QPalette* palette(Palette type = SystemPalette) const override;
diff -Naur hollywood-desktop-1.0/desktop/platformtheme/src/platformtheme.cc hollywood-desktop-1.0-qt65/desktop/platformtheme/src/platformtheme.cc
--- hollywood-desktop-1.0/desktop/platformtheme/src/platformtheme.cc	2023-03-12 18:24:41.000000000 -0700
+++ hollywood-desktop-1.0-qt65/desktop/platformtheme/src/platformtheme.cc	2023-08-23 17:42:30.262454072 -0700
@@ -114,7 +114,7 @@
 
 }
 
-#if QT_VERSION > 0x060000
+#if QT_VERSION > 0x060000 && QT_VERSION < 0x060500
 QPlatformTheme::Appearance HollywoodPlatformTheme::appearance() const
 {
     //0x0001 == light, 0x0002 == dark
