diff -Naur wpewebkit-2.39.5-orig/Source/WebCore/platform/glib/UserAgentGLib.cpp wpewebkit-2.39.5/Source/WebCore/platform/glib/UserAgentGLib.cpp
--- wpewebkit-2.39.5-orig/Source/WebCore/platform/glib/UserAgentGLib.cpp	2023-02-04 12:45:40.161641385 -0800
+++ wpewebkit-2.39.5/Source/WebCore/platform/glib/UserAgentGLib.cpp	2023-02-04 12:47:31.381641385 -0800
@@ -53,7 +53,8 @@
 #else
     if (chassisType() == WTF::ChassisType::Mobile)
         return "Linux";
-    return "X11";
+    // We are not X11, we are Hollywood
+    return "Hollywood";
 #endif
 }
 
@@ -65,7 +66,8 @@
 
     struct utsname name;
     uname(&name);
-    static NeverDestroyed<const String> uaOSVersion(makeString(name.sysname, ' ', name.machine));
+    // on Hollywood we will always be x86_64 even on ARM for desktop reasons
+    static NeverDestroyed<const String> uaOSVersion(makeString(name.sysname, ' ', "x86_64"));
     return uaOSVersion;
 #else
     // We will always claim to be Safari in Intel Mac OS X, since Safari without
