diff -Naur abuild-3.11.21-orig/abuild.in abuild-3.11.21/abuild.in
--- abuild-3.11.21-orig/abuild.in	2023-09-11 11:20:54.703051695 -0700
+++ abuild-3.11.21/abuild.in	2023-09-11 11:21:11.903135395 -0700
@@ -1,4 +1,4 @@
-#!/bin/ash -e
+#!/bin/bash -e
 
 # abuild - build apk packages (light version of makepkg)
 # Copyright (c) 2008-2015 Natanael Copa <ncopa@alpinelinux.org>
@@ -27,8 +27,8 @@
 : ${APK:="apk"}
 : ${ADDUSER:="abuild-adduser"}
 : ${ADDGROUP:="abuild-addgroup"}
-: ${CC:="gcc"}
-: ${CXX:="g++"}
+: ${CC:="clang"}
+: ${CXX:="clang++"}
 : ${SETFATTR:="setfattr"}
 
 apk_opt_wait="--wait 30"
