diff -Naur abuild-3.10.0_rc1-orig/functions.sh.in abuild-3.10.0_rc1/functions.sh.in
--- abuild-3.10.0_rc1-orig/functions.sh.in	2022-11-24 20:24:55.287322521 -0800
+++ abuild-3.10.0_rc1/functions.sh.in	2022-11-24 20:25:14.437322530 -0800
@@ -30,7 +30,7 @@
 
 hostspec_to_arch() {
 	case "$1" in
-	aarch64*-*-*-*)		echo "aarch64" ;;
+	aarch64*-*-*)		echo "aarch64" ;;
 	arm*-*-*-*eabi)		echo "armel" ;;
 	armv6*-*-*-*eabihf)	echo "armhf" ;;
 	armv7*-*-*-*eabihf)	echo "armv7" ;;
