diff -Naur abuild-3.9.0-orig/functions.sh.in abuild-3.9.0/functions.sh.in
--- abuild-3.9.0-orig/functions.sh.in	2022-11-24 20:03:04.199364335 -0800
+++ abuild-3.9.0/functions.sh.in	2022-11-24 20:08:30.542405609 -0800
@@ -4,28 +4,28 @@
 program=${0##*/}
 
 arch_to_hostspec() {
-	case "$1" in
-	aarch64)	echo "aarch64-alpine-linux-musl" ;;
-	armel)		echo "armv5-alpine-linux-musleabi" ;;
-	armhf)		echo "armv6-alpine-linux-musleabihf" ;;
-	armv7)		echo "armv7-alpine-linux-musleabihf" ;;
-	loongarch32)	echo "loongarch32-alpine-linux-musl" ;;
-	loongarchx32)	echo "loongarchx32-alpine-linux-musl" ;;
-	loongarch64)	echo "loongarch64-alpine-linux-musl" ;;
-	mips)           echo "mips-alpine-linux-musl" ;;
-	mips64)         echo "mips64-alpine-linux-musl" ;;
-	mipsel)         echo "mipsel-alpine-linux-musl" ;;
-	mips64el)       echo "mips64el-alpine-linux-musl" ;;
-	ppc)		echo "powerpc-alpine-linux-musl" ;;
-	ppc64)		echo "powerpc64-alpine-linux-musl" ;;
-	ppc64le)	echo "powerpc64le-alpine-linux-musl" ;;
-	riscv32)	echo "riscv32-alpine-linux-musl" ;;
-	riscv64)	echo "riscv64-alpine-linux-musl" ;;
-	s390x)		echo "s390x-alpine-linux-musl" ;;
-	x86)		echo "i586-alpine-linux-musl" ;;
-	x86_64)		echo "x86_64-alpine-linux-musl" ;;
-	*)		echo "unknown" ;;
-	esac
+        case "$1" in
+        aarch64)        echo "aarch64-linux-gnu" ;;
+        armel)          echo "armv5-pc-linux-gnueabi" ;;
+        armhf)          echo "armv6-pc-linux-gnueabihf" ;;
+        armv7)          echo "armv7-pc-linux-gnueabihf" ;;
+        loongarch32)    echo "loongarch32-pc-linux-gnu" ;;
+        loongarchx32)   echo "loongarchx32-pc-linux-gnu" ;;
+        loongarch64)    echo "loongarch64-pc-linux-gnu" ;;
+        mips)           echo "mips-pc-linux-gnu" ;;
+        mips64)         echo "mips64-pc-linux-gnu" ;;
+        mipsel)         echo "mipsel-pc-linux-gnu" ;;
+        mips64el)       echo "mips64el-pc-linux-gnu" ;;
+        ppc)            echo "powerpc-pc-linux-gnu" ;;
+        ppc64)          echo "powerpc64-pc-linux-gnu" ;;
+        ppc64le)        echo "powerpc64le-pc-linux-gnu" ;;
+        riscv32)        echo "riscv32-pc-linux-gnu" ;;
+        riscv64)        echo "riscv64-pc-linux-gnu" ;;
+        s390x)          echo "s390x-pc-linux-gnu" ;;
+        x86)            echo "i586-pc-linux-gnu" ;;
+        x86_64)         echo "x86_64-pc-linux-gnu" ;;
+        *)              echo "unknown" ;;
+        esac
 }
 
 hostspec_to_arch() {
