diff -Naur libapk-qt-v0.4.4-orig/CMakeLists.txt libapk-qt-v0.4.4/CMakeLists.txt
--- libapk-qt-v0.4.4-orig/CMakeLists.txt	2023-10-05 20:08:55.660000000 -0700
+++ libapk-qt-v0.4.4/CMakeLists.txt	2023-10-05 20:09:27.390000000 -0700
@@ -35,11 +35,11 @@
 set(CMAKE_AUTOMOC ON)
 
 # Required by ApkQt
-set(QT_DEFAULT_MAJOR_VERSION 5)
+set(QT_DEFAULT_MAJOR_VERSION 6)
 find_package(QT NAMES Qt5 Qt6 REQUIRED COMPONENTS Core)
 message(STATUS "Qt major version: ${QT_VERSION_MAJOR}")
-find_package(Qt${QT_VERSION_MAJOR} REQUIRED COMPONENTS Core)
-#find_package(Qt6 REQUIRED COMPONENTS Core) # Uncomment this instead of line above to try Qt6
+#find_package(Qt${QT_VERSION_MAJOR} REQUIRED COMPONENTS Core)
+find_package(Qt6 REQUIRED COMPONENTS Core) # Uncomment this instead of line above to try Qt6
 find_package(LibApk REQUIRED)
 
 if(QT_MAJOR_VERSION EQUAL "6")
diff -Naur libapk-qt-v0.4.4-orig/examples/CMakeLists.txt libapk-qt-v0.4.4/examples/CMakeLists.txt
--- libapk-qt-v0.4.4-orig/examples/CMakeLists.txt	2023-10-05 20:08:55.660000000 -0700
+++ libapk-qt-v0.4.4/examples/CMakeLists.txt	2023-10-05 20:10:26.450000000 -0700
@@ -13,7 +13,7 @@
 
 include(FeatureSummary)
 
-find_package(Qt5 CONFIG REQUIRED COMPONENTS Core)
+find_package(Qt6 CONFIG REQUIRED COMPONENTS Core)
 find_package(ApkQt CONFIG REQUIRED)
 
 add_executable(test_ApkQt
@@ -21,7 +21,7 @@
 )
 
 target_link_libraries(test_ApkQt PRIVATE
-    Qt5::Core
+    Qt6::Core
     apk-qt
 )
 
diff -Naur libapk-qt-v0.4.4-orig/src/CMakeLists.txt libapk-qt-v0.4.4/src/CMakeLists.txt
--- libapk-qt-v0.4.4-orig/src/CMakeLists.txt	2023-10-05 20:08:55.670000000 -0700
+++ libapk-qt-v0.4.4/src/CMakeLists.txt	2023-10-05 20:09:52.130000000 -0700
@@ -49,7 +49,7 @@
 )
 
 
-target_link_libraries(apk-qt PUBLIC Qt::Core)
+target_link_libraries(apk-qt PUBLIC Qt6::Core)
 
 if (USE_STATIC_LIBAPK)
     target_link_libraries(apk-qt PRIVATE LibApk::LibApkStatic)
diff -Naur libapk-qt-v0.4.4-orig/tests/CMakeLists.txt libapk-qt-v0.4.4/tests/CMakeLists.txt
--- libapk-qt-v0.4.4-orig/tests/CMakeLists.txt	2023-10-05 20:08:55.680000000 -0700
+++ libapk-qt-v0.4.4/tests/CMakeLists.txt	2023-10-05 20:10:07.670000000 -0700
@@ -7,37 +7,37 @@
 )
 
 add_executable(test_openclose test_openclose.cpp)
-target_link_libraries(test_openclose apk-qt Qt5::Core)
+target_link_libraries(test_openclose apk-qt Qt6::Core)
 
 add_executable(test_db_update test_db_update.cpp)
-target_link_libraries(test_db_update apk-qt Qt5::Core)
+target_link_libraries(test_db_update apk-qt Qt6::Core)
 
 add_executable(test_fakeroot test_fakeroot.cpp)
-target_link_libraries(test_fakeroot apk-qt Qt5::Core)
+target_link_libraries(test_fakeroot apk-qt Qt6::Core)
 
 add_executable(test_add test_add.cpp)
-target_link_libraries(test_add apk-qt Qt5::Core)
+target_link_libraries(test_add apk-qt Qt6::Core)
 
 add_executable(test_del test_del.cpp)
-target_link_libraries(test_del apk-qt Qt5::Core)
+target_link_libraries(test_del apk-qt Qt6::Core)
 
 add_executable(test_print_packages test_print_packages.cpp)
-target_link_libraries(test_print_packages apk-qt Qt5::Core)
+target_link_libraries(test_print_packages apk-qt Qt6::Core)
 
 add_executable(test_changeset test_changeset.cpp)
-target_link_libraries(test_changeset apk-qt Qt5::Core)
+target_link_libraries(test_changeset apk-qt Qt6::Core)
 
 add_executable(test_progressfd test_progressfd.cpp)
-target_link_libraries(test_progressfd apk-qt Qt5::Core)
+target_link_libraries(test_progressfd apk-qt Qt6::Core)
 
 add_executable(test_async_api test_async_api.cpp)
-target_link_libraries(test_async_api apk-qt Qt5::Core)
+target_link_libraries(test_async_api apk-qt Qt6::Core)
 
 add_executable(test_async_api_upgrade test_async_api_upgrade.cpp)
-target_link_libraries(test_async_api_upgrade apk-qt Qt5::Core)
+target_link_libraries(test_async_api_upgrade apk-qt Qt6::Core)
 
 add_executable(test_reposconfig test_reposconfig.cpp)
-target_link_libraries(test_reposconfig apk-qt Qt5::Core)
+target_link_libraries(test_reposconfig apk-qt Qt6::Core)
 
 ###################################
 # Tests are executed in order, so:
