diff -Naur clang-18.1.8.src-orig/lib/Driver/ToolChains/Gnu.cpp clang-18.1.8.src/lib/Driver/ToolChains/Gnu.cpp
--- clang-18.1.8.src-orig/lib/Driver/ToolChains/Gnu.cpp	2024-08-05 18:35:59.298562773 +0000
+++ clang-18.1.8.src/lib/Driver/ToolChains/Gnu.cpp	2024-08-05 18:37:05.218847733 +0000
@@ -2482,15 +2482,15 @@
       "x86_64-redhat-linux",    "x86_64-suse-linux",
       "x86_64-manbo-linux-gnu", "x86_64-linux-gnu",
       "x86_64-slackware-linux", "x86_64-unknown-linux",
-      "x86_64-amazon-linux"};
+      "x86_64-amazon-linux",	"x86_64-originull-linux-gnu"};
   static const char *const X32Triples[] = {"x86_64-linux-gnux32",
                                            "x86_64-pc-linux-gnux32"};
   static const char *const X32LibDirs[] = {"/libx32", "/lib"};
-  static const char *const X86LibDirs[] = {"/lib32", "/lib"};
+  static const char *const X86LibDirs[] = {"/usr/lib32", "/lib"};
   static const char *const X86Triples[] = {
       "i586-linux-gnu",      "i686-linux-gnu",        "i686-pc-linux-gnu",
       "i386-redhat-linux6E", "i686-redhat-linux",     "i386-redhat-linux",
-      "i586-suse-linux",     "i686-montavista-linux",
+      "i586-suse-linux",     "i686-montavista-linux", "i686-originull-linux-gnu"
   };
 
   static const char *const LoongArch64LibDirs[] = {"/lib64", "/lib"};
